src/hotspot/share/opto/addnode.hpp

```diff
@@ -253,6 +253,38 @@ class XorLNode : public AddNode {
   virtual uint ideal_reg() const { return Op_RegL; }
 };
 
+template<class S, class U>
+static S calc_xor_max(const S hi_0, const S hi_1) {
```

Comment:
If this is exposed in the header file, then I think it would be better to have it be a static function of a class to avoid convoluting the global namespace. Suggestion: Can you have `XorINode::calc_max` which will call a static `calc_xor_max` inside the cpp file.

Reply:
I've reorganized it.